tests/base/parameterframe/test_parameterframe.py

```diff
@@ -253,6 +254,140 @@ def test_tabulate_method_columns_have_correct_data(self, head_keys):
                         raise ve
 
 
+@dataclass
+class UnitFrame1(ParameterFrame):
+    length: Parameter[float]
+    time: Parameter[int]
+    mass: Parameter[int]
+
+
+@dataclass
+class UnitFrame2(ParameterFrame):
+    magfield: Parameter[float]
+    energy: Parameter[float]
+    pressure: Parameter[int]
+
+
+@dataclass
+class UnitFrame3(ParameterFrame):
+    damagepertime: Parameter[float]
+    timeperdamage: Parameter[float]
+    damage: Parameter[float]
+    perdamage: Parameter[float]
+    on_time: Parameter[float]
+    per_on_time: Parameter[float]
+
+
+@dataclass
+class UnitFrame4(ParameterFrame):
+    angle: Parameter[float]
+    angle2: Parameter[float]
+    angleperthing: Parameter[float]
+    thingperangle: Parameter[float]
+
+
+@dataclass
+class UnitFrame5(ParameterFrame):
+    wtf1: Parameter[float]
+    wtf2: Parameter[float]
+    wtf3: Parameter[float]
+
+
+class TestParameterFrameUnits:
```

Comment:
Do we have a/some test(s) for unknown units? E.g., `{"value": 180.5, "unit": "not_a_known_unit"}`. What behaviour do we want? Maybe a warning?

Reply:
Could be a warning and conversion to dimensionless. Or could just error. I havent taken into account if the unit is the correct for that variable which I dont know might need to be coded somewhere.

Reply:
Discussed offline: throw an error if `pint` does not recognise the unit, but allow any units `pint` recognises, even if we don't have a 'standard' unit conversion for it.